### DISCUSSION TITLE
Insert pictures with a pre-defined size

### DISCUSSION THREAD

**Discussion by tymmek** [zim-desktop-wiki/zim-desktop-wiki#2009] [Ideas]
Hello, 

I was wondering whether in Zim one can set a **"default" picture size** upon insertion via drag-n-drop / insert menu methods? For example for all inserted pictures their visible height in the text is set to 100 px while their actual size is left unchanged (no file modification). I usually manually edit the pictures as links such that by clicking on them they open to their full resolution via the standard file viewer. Most of the time I want to insert pictures as thumbnails, so in the few cases that I don't, I can use the editor.

Could this be done by adding predefined image sizes to `style.conf` file?  Better to implement as a plugin?
I understand that the file drawer provides thumbnail capabilities, however I like the ability to see pictures next to their descriptions.

Obviously (while on topic) there are many ways to streamline the experience:
- A popup menu appearing during drag-n-drop with editable options such as
  - "Insert as thumbnail (height=100px)",
  - "Insert actual size",
  - "Insert as link",
  - "Other (open editor)"
- embedded resizing controls appearing on the image when hovering with the mouse (similar to the Zim table editor)

**Comment by introt** [comment]
Hi! Such a feature doesn't currently exist, please see https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1937#discussioncomment-2205475 for links to related open tickets.

**Reply by tymmek** [reply]
Thank you!
